packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -306,6 +321,33 @@ EmbedFlutterFrameworks() {
     RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/App.framework/App"
     RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/Flutter.framework/Flutter"
   fi
+
+  AddObservatoryBonjourService
+}
+
+# Add the observatory publisher Bonjour service to the produced app bundle Info.plist.
+AddObservatoryBonjourService() {
+  ParseFlutterBuildMode
+  local build_mode="${retval}"
+  # Debug and profile only.
+  if [[ "${build_mode}" == "release" ]]; then
+    return
+  fi
+  local built_products_plist="${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}"
+
+  # If there are already NSBonjourServices specified by the app (uncommon), insert the observatory service name to the existing list.
+  if plutil -extract NSBonjourServices xml1 -o - "${built_products_plist}"; then
+    RunCommand plutil -insert NSBonjourServices.0 -string "_dartobservatory._tcp" "${built_products_plist}"
+  else
+    # Otherwise, add the NSBonjourServices key and observatory service name.
+    RunCommand plutil -insert NSBonjourServices -json "[\"_dartobservatory._tcp\"]" "${built_products_plist}"
+  fi
+
+  # Don't override the local network description the Flutter app developer specified (uncommon).
+  # This text will appear below the "Your app would like to find and connect to devices on your local network" permissions popup.
+  if ! plutil -extract NSLocalNetworkUsageDescription xml1 -o - "${built_products_plist}"; then
+    RunCommand plutil -insert NSLocalNetworkUsageDescription -string "Allow Flutter tools on your computer to connect and debug your Flutter application. This prompt will not appear on release builds." "${built_products_plist}"
```

Comment:
Check out the screenshot in the description.  We can probably remove the second "Flutter":
> Allow Flutter tools on your computer to connect and debug your application. This prompt will not appear on release builds.

Reply:
On the second connect. I'd maybe leave it. Otherwise people might think I'm just flutter running and then maybe taking a screenshot or something. I'm not technically "debugging", why doesn't it work?

On the second Flutter, don't have a strong opinion. To be specific, we're not really debugging their (objective-c) application. This is strictly dart/observatory/flutter stuff.